scylla/src/transport/connection.rs

```diff
@@ -649,14 +696,25 @@ impl Connection {
                 )))
             })?;
 
+        // Dropping `notifier` (before calling `notifier.disable()`) will send a notification to
+        // `Connection::router`. This notification is then used to mark a `stream_id` associated
+        // with this request as orphaned and free associated resources.
```

Comment:
Isn't the following scenario possible?
1. The runtime yields from this code right after submitting the request above (`self.submit_channel`) succeeded, but before the line below runs (`let notifier = ...`).
2. This code's caller wrapped it in tokio::timeout, which fires
3. The request was already sent with an allocated stream id, but it's not going to be orphaned when no response arrives, because we yielded before creating the notifier

Reply:
Notifier is created in the `Connection::writer`, and then send here, to `Connection::send_request`. If a send fails, the notifier is dropped on the `Connection::writer` side, which also triggers mechanism for stream id orphaning.

So the third point is invalid, the allocated stream id will be marked as orphaned.